### PR TITLE
Trigger an error if there is an incomplete entry at the end of stream

### LINKED ISSUE #82: Unexpected end of data should raise error when extracting
When the input data ends while tar-stream waits for more data to extract, it doesn't raise an error. This means that a truncated tar file will extract without reporting errors, while creating incomplete files. 

If there is still missing data for a file or a partially read header, an error should be raised instead.

This issue is likely also the cause for #71. If only a short file is processed (shorter than a tar header), no errors are raised since the partial header data is never processed.

### PR DESCRIPTION
**Pull request by sth**
This is a pull request that fixes #82.

It keeps track if there is an entry currently being parsed, and triggers an error if that's still the case when `_final()` is called.

`_final()` was introduced in node v8, but since tar-stream uses `stream-readable` instead of the standard library streams, it works for all node versions.

### CODE REVIEW

**Comment by mafintosh** [comment]
Node version should not matter as we use the readable-stream module.

**Comment by mafintosh** [comment]
Ah you said that already 😁

File: extract.js [hunk 1]
```diff
@@ -196,10 +197,11 @@ Extract.prototype.destroy = function (err) {
   if (this._stream) this._stream.emit('close')
 }
 
-Extract.prototype._parse = function (size, onparse) {
+Extract.prototype._parse = function (size, onparse, newentry) {
   if (this._destroyed) return
   this._offset += size
   this._missing = size
+  if (newentry) this._partial = false
```
Comment (mafintosh) [inline extract.js]: Couldn't you just do `this._partial = onparse !== onheader` and not have the extra arg?

**Comment by sth** [comment]
I removed the extra arg, it didn't really make things clearer.

**Pull request merged**